packages/core/strapi/lib/types/core/config/database.d.ts

```diff
@@ -0,0 +1,63 @@
+import { Utils } from '../..';
+
+type Client = 'mysql' | 'postgres' | 'sqlite';
+
+type SSLConfig = {
+  rejectUnauthorized?: boolean;
+  key?: string;
+  cert?: string;
+  ca?: string;
+  capath?: string;
+  cipher?: string;
+};
+
+type PoolConfig = {
+  min?: number;
+  max?: number;
+  acquireTimeoutMillis?: number;
+  createTimeoutMillis?: number;
+  destroyTimeoutMillis?: number;
+  idleTimeoutMillis?: number;
+  reapIntervalMillis?: number;
+  createRetryIntervalMillis?: number;
+  // Todo: add types for these callbacks
+  afterCreate?: (conn: any, done: (err?: Error, conn?: any) => void) => void;
+};
+
+type Connection<TClient extends Client> = {
+  database: string;
+  user: string;
+  password: string;
+  port: number;
+  host: string;
+  ssl?: SSLConfig | boolean;
+  connectionString?: string;
+  timezone?: string;
+} & { [key: string]: unknown } & Utils.Expression.MatchFirst<
+    [[Utils.Expression.StrictEqual<TClient, 'postgres'>, { schema?: string }]],
+    {}
+  >;
+
+type SqliteConnection = {
+  filename: string;
+} & { [key: string]: unknown };
+
+export interface DBConfig<TClient extends Client> {
+  connection: {
+    client: TClient;
+    connection: Utils.Expression.MatchFirst<
+      [[Utils.Expression.StrictEqual<TClient, 'sqlite'>, SqliteConnection]],
+      Connection<TClient>
+    >;
```

Comment:
I'm wondering if there is a cleaner way to write this conditions, the only thing I can come up with is, but maybe @Convly  has an opinion on this, since it's something that is being repeated all over the file:
```suggestion
    connection: Utils.Expression.If<
      Utils.Expression.Extends<TClient, 'sqlite'>,
      SqliteConnection,
      Connection<TClient>
    >;
```

Reply:
The If looks more appropriate than the MatchFirst for those conditions indeed.